### PR TITLE
Add feature tests for search faceting

### PR DESCRIPTION
**Pull request by ConnorSheremeta**
- Added tests for all facets
- Added test for clicking more to view all facets
- Added tests for using more facet modal
#1622

### CODE REVIEW

File: spec/features/search_faceting_spec.rb [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    expect(page).to have_text('Acting')
+    expect(page).to have_no_text('Environnement')
+    expect(page).to have_no_text('Monologues')
+    within('.blacklight-subject_topic_facet') do
+      click_on 'more'
+    end
+
+    expect(page).to have_text('Human Ecology')
+    click_on 'A-Z Sort'
+    expect(page).to have_no_text('Human Ecology')
```
Comment (pgwillia) [inline spec/features/search_faceting_spec.rb]: This failed for me 
```
Failures:

  1) Search faceting User can utilize the more modal
     Failure/Error: expect(page).to have_no_text('Human Ecology')
       expected #has_no_text?("Human Ecology") to return true, got false
     # ./spec/features/search_faceting_spec.rb:36:in `block (2 levels) in <top (required)>'
```

**Comment by ConnorSheremeta** [comment]
After some digging I am unable to find a way to remove the ingest noise, it would be nice if every added record didn't output it's own line!

File: spec/features/search_faceting_spec.rb (hunk 1 above)
Comment (ConnorSheremeta) [reply spec/features/search_faceting_spec.rb]: Hmm, strange. I wonder why this is passing on my system and CI but not on yours, would you mind posting a screenshot at this point with screenshot_and_save_page? I'm curious if it's just not clicking on 'A-Z Sort' for whatever reason

**Comment by pgwillia** [comment]
> After some digging I am unable to find a way to remove the ingest noise, it would be nice if every added record didn't output it's own line!

Yeah, I've dug at various points too. Thanks for looking into it.  I'll open an issue so that we can think about it some more.

File: spec/features/search_faceting_spec.rb (hunk 1 above)
Comment (pgwillia) [reply spec/features/search_faceting_spec.rb]: Weird.  Not failing anymore, but the coverage has gone down? `971 / 1214 LOC (79.98%) covered.`

Comment (pgwillia) [reply spec/features/search_faceting_spec.rb]: Travis caught this one too https://api.travis-ci.org/v3/job/531708540/log.txt :(
```
Failures:

  1) Search faceting User can utilize the more modal
     [31mFailure/Error: [0m[32mexpect[0m(page).to have_no_text([31m[1;31m'[0m[31mHuman Ecology[1;31m'[0m[31m[0m)[0m
     [31m  expected #has_no_text?("Human Ecology") to return true, got false[0m
     [36m# ./spec/features/search_faceting_spec.rb:36:in `block (2 levels) in <top (required)>'[0m
```

Comment (ConnorSheremeta) [reply spec/features/search_faceting_spec.rb]: Hmm the log for that job is showing me that there were no failures?

Comment (pgwillia) [reply spec/features/search_faceting_spec.rb]: My bad.  I guess when I 'restarted job' it was over-written.  I have a feeling that this one will come back.

**Pull request merged**